### PR TITLE
Fixed ES-36, ES-64

### PR DESCRIPTION
**Pull request by HimajaDhanyamraju2**

### CODE REVIEW

File: mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/validator/IdentityDataValidator.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.mosip.esignet.mock.identitysystem.dto.IdentityData;
+
+@Component
+public class IdentityDataValidator implements ConstraintValidator<IdData, IdentityData> {
+
+	@Value("${mosip.esignet.mock.supported-fields}")
+	private String supportedFields;
```
Comment (vishwa-vyom) [inline mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/validator/IdentityDataValidator.java:23]: There is a way in @Value to get directly list of string, lets use that

File: mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/validator/IdentityDataValidator.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+	@Value("${mosip.esignet.mock.number-of-extra-fields:10}")
+	private String extraFields;
```
Comment (vishwa-vyom) [inline mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/validator/IdentityDataValidator.java:26]: Lets declare this as int or long directly and can we rename to allowedExtraFieldsCount ?

File: mock-identity-system/src/main/resources/messages.properties [hunk 3]
```diff
@@ -8,7 +8,9 @@ invalid_phone=Invalid Phone number
 invalid_encoded_photo=Invalid encoded photo
 invalid_postalcode=Invalid postal code
 invalid_date_of_birth=Invalid Date of birth
-invalid_fullname=Invalid fullName
+invalid_fullname=Invalid Full Name
+invalid_firstname=Invalid First Name
+invalid_lastname=Invalid Last Name
 invalid_gender=Invalid gender
 invalid_street_address=Invalid street address
 invalid_locality=Invalid Locality
```
Comment (vishwa-vyom) [inline mock-identity-system/src/main/resources/messages.properties:16]: I am not able to find the code change in 
https://github.com/mosip/esignet-mock-services/blob/develop/mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/service/impl/AuthenticationServiceImpl.java#L262

where we should send the first name and last name if consented by the user ?

File: mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/validator/IdentityDataValidator.java (hunk 2 above)
Comment (vishwa-vyom) [inline mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/validator/IdentityDataValidator.java:26]: The requirement on extraFields was allowing someone to include their custom fields while creating the identity, how are we handling that if that fileds not declared in the Identity data class ?

File: mock-identity-system/src/main/resources/messages.properties (hunk 3 above)
Comment (vishwa-vyom) [reply mock-identity-system/src/main/resources/messages.properties:16]: We should also support all the standard claims here

**Review by vishwa-vyom — CHANGES_REQUESTED** [review]

File: mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/validator/IdentityDataValidator.java (hunk 1 above)
Comment (HimajaDhanyamraju2) [reply mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/validator/IdentityDataValidator.java]: Modified as suggested.

File: mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/validator/IdentityDataValidator.java (hunk 2 above)
Comment (HimajaDhanyamraju2) [reply mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/validator/IdentityDataValidator.java]: Removed the count as we discussed.

Comment (HimajaDhanyamraju2) [reply mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/validator/IdentityDataValidator.java]: Modified the approach as discussed.

**Review by vishwa-vyom — CHANGES_REQUESTED** [review]

File: mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/dto/IdentityData.java [hunk 4]
```diff
@@ -62,10 +75,19 @@ public class IdentityData {
...
-	@NotBlank(message = ErrorConstants.INVALID_EMAIL)
+	@Size(min = 1, message = ErrorConstants.INVALID_EMAIL)
 	String email;
 
-	@NotBlank(message = ErrorConstants.INVALID_PHONE)
+	@Size(min = 1, message = ErrorConstants.INVALID_PHONE)
 	String phone;
+	
+	@Size(min = 1, message = ErrorConstants.INVALID_WEBSITE)
+	String website;
```
[the remark this answers is not in the document]
Comment (HimajaDhanyamraju2) [reply mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/dto/IdentityData.java]: Removed.

**Review by vishwa-vyom — APPROVED** [review]

**Pull request merged**